v03_pipeline/lib/reference_data/clinvar.py

```diff
@@ -150,3 +152,37 @@ def _parse_clinvar_release_date(local_vcf_path: str) -> str:
                 return None
 
     return None
+
+
+def download_and_import_clinvar_txt_file(url: str, types: dict, hl_filter: str | None):
+    with tempfile.NamedTemporaryFile(
+        suffix='.txt.gz',
+        delete=False,
+    ) as tmp_file:
+        urllib.request.urlretrieve(url, tmp_file.name)  # noqa: S310
+        return hl.import_table(
+            tmp_file.name,
+            force=True,
+            filter=hl_filter,
+            impute=True,
+            types=types,
+            missing='-',
+            min_partitions=3,  # recommended 2-4 partitions per core
+        )
+
+
+def download_and_import_clinvar_submission_summary() -> hl.Table:
+    # https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/README - submission_summary.txt
+    logger.info('Getting clinvar submission summary')
+    ht = download_and_import_clinvar_txt_file(
+        url=CLINVAR_SUBMISSION_SUMMARY_URL,
+        types={'#VariationID': hl.tstr},
+        hl_filter='^(#[^:]*:|^##).*$',  # removes all comments except for the header line
```

Comment:
Would we be able to use the `comment` arg instead of a regex?

Reply:
I tried a bunch of combinations of different `comment` and `filter` and even `find_replace` to keep all of the comments except for the one with the column names (also starts with # which made it difficult). I even checked out the source code for `import_table` to see what order the params were being used.

This filter regex was the best solution I found to filter out the comments but keep the column names. I guess it's not resilient to any change in the formatting of the file comments though 

here's what the header looks like:
```
##Overview of interpretation, phenotypes, observations, and methods reported in each current submission 
##Explanation of the columns in this report
#VariationID:                   the identifier assigned by ClinVar and used to build the URL, namely https://ncbi.nlm.nih.gov/clinvar/VariationID
#ClinicalSignificance:          the germline classification on this submitted record
#DateLastEvaluated:             the last date the classification on this record was evaluated by the submitter
#Description:                   an optional free text description comment describing the rationale for the classification
#SubmittedPhenotypeInfo:        the name(s) or identifier(s) submitted as the condition for the classification 
#ReportedPhenotypeInfo:         the MedGen identifier/name combinations that the submitted condition for the classification maps to. 'na' means there is no public identifer in MedGen for the condition.
#ReviewStatus:                  the level of review for this submitted record; see http//www.ncbi.nlm.nih.gov/clinvar/docs/variation_report/#review_status
#CollectionMethod:              the method by which the submitter collected the data for the classification; see https://www.ncbi.nlm.nih.gov/clinvar/docs/spreadsheet/#collection
#OriginCounts:                  the allele origin reported by the submitter and the number of observations for each origin. ‘na’ means that the number of observations was not provided by the submitter.
#Submitter:                     the submitter of this record
#SCV:                            the accession and current version assigned by ClinVar to this submitted record
#SubmittedGeneSymbol:           the gene symbol reported in this submitted record
#ExplanationOfInterpretation:   more details if the germline classification (ClinicalSignificance) is 'other' or 'drug response'
#SomaticClinicalImpact:         the somatic classification of clinical impact on this submitted record
#Oncogenicity:                  the somatic classification of oncogenicity on this submitted record
#VariationID	ClinicalSignificance	DateLastEvaluated	Description	SubmittedPhenotypeInfo	ReportedPhenotypeInfo	ReviewStatus	CollectionMethod	OriginCounts	Submitter	SCV	SubmittedGeneSymbol	ExplanationOfInterpretation	SomaticClinicalImpact	Oncogenicity
2	Pathogenic	-	-	OMIM:613647	C3150901:Hereditary spastic paraplegia 48	criteria provided, single submitter	clinical testing	unknown:2	Paris Brain Institute, Inserm - ICM	SCV001451119.1	-	-	-	-
2	Pathogenic	Jun 29, 2010	-	SPASTIC PARAPLEGIA 48, AUTOSOMAL RECESSIVE	C3150901:Hereditary spastic paraplegia 48	no assertion criteria provided	literature only	germline:na	OMIM	SCV000020155.3	AP5Z1	-
```

Reply:
👍 makes sense to me!  thanks for the explanation!